frontend/packages/dev-console/src/components/edit-deployment/EditDeploymentPage.tsx

```diff
@@ -4,34 +4,52 @@ import { useTranslation } from 'react-i18next';
 import { RouteComponentProps } from 'react-router-dom';
 import { StatusBox } from '@console/internal/components/utils';
 import { useK8sWatchResource } from '@console/internal/components/utils/k8s-watch-hook';
-import { K8sResourceKind } from '@console/internal/module/k8s';
+import { DeploymentKind } from '@console/internal/module/k8s';
 import NamespacedPage, { NamespacedPageVariants } from '../NamespacedPage';
 import EditDeployment from './EditDeployment';
+import { getDefaultDeployment } from './utils/create-deployment-utils';
 
 export type EditDeploymentPageProps = RouteComponentProps<{ ns?: string }>;
 
 const EditDeploymentPage: React.FC<EditDeploymentPageProps> = ({ match, location }) => {
   const { t } = useTranslation();
   const namespace = match.params.ns;
   const queryParams = new URLSearchParams(location.search);
-  const kind = queryParams.get('kind');
   const name = queryParams.get('name');
-  const heading = t('devconsole~Edit {{kind}}', { kind });
-  const label = t('devconsole~Edit {{kind}} form', { kind });
+  const isNew = !name || name === '~new';
+  const kind = queryParams.get('kind');
+
+  const [watchedDeployment, loaded, loadError] = useK8sWatchResource<DeploymentKind>(
+    isNew
+      ? null
+      : {
+          kind,
+          name,
+          namespace,
+        },
+  );
+
+  const deployment: DeploymentKind = isNew ? getDefaultDeployment(namespace) : watchedDeployment;
```

Comment:
I think here or somewhere here the kind needs to be checked. At the moment, when creating a DeploymentConfig it initializes it with a Deployment.

I can see that when switching to YAML.

Also the form content differs in both cases. The Deployment Strategy for Deployments shows just

1. Recreate         and
2. Rolling Update

For DeploymentConfig it should show:

1. Recreate
2. Rolling              and
3. Custom

Reply:
@jerolimov This has been fixed now. Both the deployment and the deployment config form are working fine.